### PR TITLE
[BD-03] [TNL-7352] Initial forum structure

### PR DESCRIPTION
**Pull request by xitij2000**
Adds the initial basic structure for the discussions MFE.

**JIRA tickets**: TNL-7352

**Screenshots**: NA. Adds basic component structure that doesn't try to match visuals or functionality. 

**Sandbox URL**: No sandbox is available for this stage. 

**Merge deadline**: None

**Testing instructions**:

NA

**Reviewers**
- [x] @0x9a (initially reviewed here: https://github.com/open-craft/frontend-app-discussions.old/pull/1)
- [ ] @stvstnfrd

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @xitij2000! I've created [OSPR-4895](https://openedx.atlassian.net/browse/OSPR-4895) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

Please let us know once your PR is ready for our review and all tests are green.

**Comment by xitij2000** [comment]
@stvstnfrd  This is the base forum structure in which other PRs will be merged. It has the very basics working, but a lot of changes are coming in future PRs. This is based on a slightly older design, but that will be rectified in the rest of the PRs.

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+coverage/*
+dist/
+node_modules/
+jest.config.js
```
Comment (stvstnfrd) [inline .eslintignore:4]: Nit: Can you add trailing newline here?

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+const { createConfig } = require('@edx/frontend-build');
+
+module.exports = createConfig('eslint');
```
Comment (stvstnfrd) [inline .eslintrc.js:3]: Nit: Can you add trailing newline here?

**Review by stvstnfrd — APPROVED** [review]
Two nitpicks, but otherwise nothing controversial, :+1:

File: .env.development [hunk 3]
```diff
@@ -0,0 +1,17 @@
+NODE_ENV='development'
+PORT=2002
```
Comment (xitij2000) [inline .env.development:2]: @stvstnfrd  I picked this port as learning is 2000, and authoring is 2001 so I just picked the next one.

File: .env.test [hunk 4]
```diff
@@ -0,0 +1,15 @@
+ACCESS_TOKEN_COOKIE_NAME='edx-jwt-cookie-header-payload'
+BASE_URL='localhost:1995'
```
[the remark this answers is not in the document]
Comment (stvstnfrd) [reply .env.test:2]: It also looks like `PORT` isn't defined in this file either.
Are there other settings missing? Was this intentional or an issue w/ the template/cookiecutter?

**Review by stvstnfrd — CHANGES_REQUESTED** [review]

File: .env.test (hunk 4 above)
[the remark this answers is not in the document]
Comment (xitij2000) [reply .env.test:2]: I have not touched this file, so it's probably because of the template.

**Comment by xitij2000** [comment]
@stvstnfrd If this seems OK to you could you give it a thumbs-up so I can merge it?

**Review by 0x29a — APPROVED** [review]
:+1:

- [x] I tested that new discussions frontend loads fine and has basic structure, like forums and threads.
- [x] I read through the code
- [ ] ~I checked for accessibility issues~
- [ ] ~Includes documentation~
- [ ] ~I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.~

**Pull request merged**